service/matching/matcher_test.go

```diff
@@ -359,6 +359,26 @@ func (t *MatcherTestSuite) TestSyncMatchFailure() {
 	t.False(syncMatch)
 }
 
+func (t *MatcherTestSuite) TestQueryNoPoller() {
+	task := newInternalQueryTask(uuid.New(), &matchingservice.QueryWorkflowRequest{})
+
+	t.client.EXPECT().QueryWorkflow(gomock.Any(), gomock.Any(), gomock.Any()).Do(
+		func(ctx context.Context, req *matchingservice.QueryWorkflowRequest, arg2 ...interface{}) {
+			time.Sleep(10 * time.Millisecond)
+			task.forwardedFrom = req.GetForwardedSource()
+			resp, err := t.rootMatcher.OfferQuery(ctx, task)
+			t.Nil(resp)
+			t.Assert().Error(err, errNoRecentPoller)
```

Comment:
I can't quite put together why this test works since the default timeout is 20s?

Reply:
it was working because there was no pollers at all. Plz note that the 20s is not the timeout, it's the window in which we expect to see a poller.